### ISSUE TITLE
GHC version match check seems to be checking the version that Cabal was compiled with

### ISSUE THREAD

**Issue by adamse** [alexwl/haskell-code-explorer#35]
I'm not 100% but it seems that `getPackageGhcVersion` is actually parsing out the GHC version that cabal-install/Cabal was built with, not the GHC version for the package.

Here's a snippet of the `setup-config` I have:

```
Saved package config for shake-0.17.8 written by Cabal-2.4.1.0 using ghc-8.4
<BLAH> exe:shake
<BLAH BLAH> /Users/adam/.stack/programs/x86_64-osx/ghc-8.6.4/bin/ghc
```

**Comment by adamse** [comment]
The setup file does contain the whole of `ghc --info` a bit further along, which includes the GHC version (with all three version components).

**Comment by adamse** [comment]
I see this was already noted in https://github.com/alexwl/haskell-code-explorer/pull/33#discussion_r271672292

**Comment by alexwl** [comment]
You are right.

I wrongly assumed that the version of GHC from the header of `setup-config` is the version of GHC that was used to build the package.

It seems that this assumption may be correct when the GHC is installed by stack (because versions of GHC and Cabal are pinned), but in general, it is wrong.

I've committed the fix: 4ad872ac715e212c2c2c86e4a1a87f9c3a09ee91

**Comment by adamse** [comment]
Thanks for fixing it. You have built an excellent tool here!
